plugins/modules/sonarr_quality_profile.py

```diff
@@ -230,58 +196,67 @@ def run_module():
                 result['id'] = 0
         module.exit_json(**result)
 
+    # Get schema
+    schema_client = sonarr.QualityProfileSchemaApi(module.api)
+    try:
+        want = schema_client.get_qualityprofile_schema()
+    except Exception as e:
+        module.fail_json('Error getting quality profile schema: %s' % to_native(e.reason), **result)
+
+    # Break apart quality groups and create a dict.
+    quality_groups_dict = OrderedDict()
+
+    for item in want.items:
+        if item.quality is None:
+            for quality in item.items:
+                quality.allowed = False
+                quality_groups_dict[quality.quality.name] = quality
+        else:
+            item.allowed = False
+            quality_groups_dict[item.quality.name] = item
+
     # Populate quality groups.
-    quality_groups = []
-    for item in module.params['quality_groups']:
+    ident = 1000
```

Comment:
this makes sense, end users won't care much about which id they're pointing to, hence selecting this automatically it's a great idea